ReactiveObjCTests/RACSignalSpec.m

```diff
@@ -1984,6 +1984,20 @@ + (void)configure:(Configuration *)configuration {
     [subject sendCompleted];
     expect(@(completed)).to(beTruthy());
   });
+	
+  qck_it(@"should dispose previous inner signal before subscribing to new inner signal", ^{
+    RACSubject *otherSignal = [RACSubject subject];
+    RACSignal *signal = [RACSignal createSignal:^RACDisposable *(id<RACSubscriber> subscriber) {
+      [otherSignal sendNext:@"foo"];
```

Comment:
minor: consider adding a test where otherSignal sends from another thread, in order to test for thread safety.

Reply:
done below